test/tests/carousel_carousel-2-tablist.js

```diff
@@ -0,0 +1,450 @@
+    );
+  }
+});
+
+ariaTest('HOME key moves focus and selects tab', exampleFile, 'key-home', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
+
+    // Put focus on the tab
+    await openTabAtIndex(t, index);
+
+    // Send the home key to the tab
+    await tabs[index].sendKeys(Key.HOME);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, 0),
+      'home key on tab "' + index + '" should put focus on the first tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, 0),
+      'home key on tab "' + index + '" should set aria-selected="true" on the first tab.'
+    );
+    t.true(
+      await tabpanels[0].isDisplayed(),
+      'home key on tab "' + index + '" should display the first tab.'
+    );
+  }
+});
+
+ariaTest('END key moves focus and selects tab', exampleFile, 'key-end', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
+
+    // Put focus on the tab
+    await openTabAtIndex(t, index);
+
+    // Send the end key to the tab
+    await tabs[index].sendKeys(Key.END);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, tabs.length - 1),
+      'home key on tab "' + index + '" should put focus on the last tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, tabs.length - 1),
+      'home key on tab "' + index + '" should set aria-selected="true" on the last tab.'
+    );
+    t.true(
+      await tabpanels[tabs.length - 1].isDisplayed(),
+      'home key on tab "' + index + '" should display the last tab.'
+    );
+  }
+});
+
+// Keyboard interaction
+
+ariaTest('TAB moves key through buttons', exampleFile, 'rotation-key-tab', async (t) => {
+  t.plan(1);
+
+  await assertTabOrder(t, ex.allFocusableItems);
+});
+
+ariaTest('ENTER pause and start carousel motion', exampleFile, 'rotation-enter-or-space-toggle', async (t) => {
+  t.plan(2);
+
+  let activeElement = await t.context.session.findElement(By.css(ex.activeCarouselItem)).getAttribute('aria-label');
+
+  await t.context.session.findElement(By.css(ex.pausePlayButtonSelector)).sendKeys(Key.ENTER);
+  // Move focus from widget
+  await t.context.session.findElement(By.css(ex.pausePlayButtonSelector)).sendKeys(Key.chord(Key.SHIFT, Key.TAB));
+
+  let compareWithNextElement = await t.context.session.wait(async function () {
+    let newActiveElement = await t.context.session.findElement(By.css(ex.activeCarouselItem)).getAttribute('aria-label');
+    return activeElement === newActiveElement;
```

Comment:
It might be worth it to manually set the `carouselTabList.timeInterval` manually to a very short interval, e.g. 100ms, then wait a specific amount of time greater than 100ms here and check that the active tab has changed. Theoreticallly, this implementation could wait until the carousel cycles all the way back to the first element, then return a false positive.

Reply:
context.session.wait actually is a selenium webdriver command which runs the command in a loop until the conditions are met or the time out occurs -- so it's pretty impossible for it to have looped around luckily!

Reply:
but here is the link to the docs for the curious: https://selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_ThenableWebDriver.html#wait